// youtube-channel-statistics/lib/syko.js
const request = require('request');

class SykoCoder {
    static async getYoutubeChannelInfo(channelID) {
        return new Promise(async (resolve, reject) => {

            try {
        const key = "AIzaSyCABpk7NHJN9kye1_Me_gyCnlTp_gtC1pI";
        const url = "https://www.googleapis.com/youtube/v3/channels?part=statistics&id=" + channelID + "&key=" + key;
        let data = []
        request({
            method: 'GET',
            url: url
        }, function (err, response, text) {
            if (err) {
        
                return;
            }
        
            var json = JSON.parse(text);
        
            let subscribes = json.items[0].statistics.subscriberCount
            let viewCountt = json.items[0].statistics.viewCount
            let videoCountt = json.items[0].statistics.videoCount

            data.push({
                subscribe: subscribes,
                viewcount: viewCountt,
                videocount: videoCountt
            })
            resolve(data)
            
        })
        } catch (error) {
            reject(error);
        }
    })
    }
}




    module.exports = SykoCoder;
